### PR TITLE
Influxdb

### PR DESCRIPTION
**Pull request by JasonIamAUnixAdmin**
First pass at installing influxdb

### CODE REVIEW

File: monitor/etc/influxdb/influxdb.conf [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+### [data]
+###
+### Controls where the actual shard data for InfluxDB lives and how it is
+### flushed from the WAL. "dir" may need to be changed to a suitable place
+### for your system, but the WAL settings are an advanced configuration. The
+### defaults should work for most systems.
+###
+
+[data]
+#  dir = "/mnt/influxdb/data"
```
Comment (cpenner461) [inline monitor/etc/influxdb/influxdb.conf]: remove this line (since it's already commented out)?

File: monitor/etc/influxdb/influxdb.conf [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  write-timeout = "5s" # The time within which a write operation must complete on the cluster.
+
+###
+### [retention]
+###
+### Controls the enforcement of retention policies for evicting old data.
+###
+
+[retention]
+#  enabled = true
```
Comment (cpenner461) [inline monitor/etc/influxdb/influxdb.conf]: remove this line (since it's already commented out)?

File: monitor/influxdb/init.sls [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+/etc/opt/influxdb/influxdb.conf:
+  file:
+    - managed
+    - source: salt://monitor/etc/influxdb/influxdb.conf
+    - template: jinja
+    - makedirs: true
+    - require:
+      - pkg: influxdb_pkg
+
+{{ storage_dir }}/data:
```
Comment (cpenner461) [inline monitor/influxdb/init.sls]: does this work?  I think you'd need something like `{{pillar.monitoring.storage_dir}}`, or `{% set storage_dir= salt['pillar.get']('monitoring:influxdb:storage_dir', '/mnt/influxdb') %}` at the top of the file.

File: monitor/etc/influxdb/influxdb.conf [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+###
+### Controls where the actual shard data for InfluxDB lives and how it is
+### flushed from the WAL. "dir" may need to be changed to a suitable place
+### for your system, but the WAL settings are an advanced configuration. The
+### defaults should work for most systems.
+###
+
+[data]
+#  dir = "/mnt/influxdb/data"
+  dir = "{{ storage_dir }}/data"
```
Comment (cpenner461) [inline monitor/etc/influxdb/influxdb.conf]: does this work?  I think you'd need something like `{{pillar.monitoring.storage_dir}}`, or `{% set storage_dir= salt['pillar.get']('monitoring:influxdb:storage_dir', '/mnt/influxdb') %}` at the top of the file.

**Comment by JasonIamAUnixAdmin** [comment]
Updated based on comments. Hard to test pillar data locally.

**Comment by cpenner461** [comment]
:+1:

**Pull request merged**